### PR TITLE
IfcMobileTelecommunicationsAppliance Rail Import BIMQ 2021-07-21T06:43

### PR DESCRIPTION
**Pull request by jmirtsch**
Rail properties for IfcMobileTelecommunicationsAppliance

### CODE REVIEW

File: IFC4x3/Properties/e/ExchangeCapacity_3BKPzAHt10s8UrVNVqgpxw/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ExchangeCapacity_3BKPzAHt10s8UrVNVqgpxw" Name="ExchangeCapacity" UniqueId="cb519f4a-4770-40d8-87b5-7d77f4ab3efa" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/e/ExchangeCapacity_3BKPzAHt10s8UrVNVqgpxw/DocProperty.xml:2]: No need for change

File: IFC4x3/Properties/m/MaximumNumberOfConnectedBSCs_2y9Cxd6R9B5xNzs7Z2VO_B/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MaximumNumberOfConnectedBSCs_2y9Cxd6R9B5xNzs7Z2VO_B" Name="MaximumNumberOfConnectedBSCs" UniqueId="bc24cee7-19b2-4b17-b5fd-d878c27d8f8b" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/m/MaximumNumberOfConnectedBSCs_2y9Cxd6R9B5xNzs7Z2VO_B/DocProperty.xml:2]: No need for change

File: IFC4x3/Properties/m/MaximumNumberOfRRUs_0tAsowuCH8xgtZ_ZgM$4Ug/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="MaximumNumberOfRRUs_0tAsowuCH8xgtZ_ZgM4Ug" Name="MaximumNumberOfRRUs" UniqueId="372b6cba-e0c4-48ee-ade3-fa3a96fc47aa" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/m/MaximumNumberOfRRUs_0tAsowuCH8xgtZ_ZgM$4Ug/DocProperty.xml:2]: No need for change

File: IFC4x3/Properties/n/NumberOfAntennas_39HzNd1Dv8guqqzDmmmPhJ/DocProperty.xml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfAntennas_39HzNd1Dv8guqqzDmmmPhJ" Name="NumberOfAntennas" UniqueId="c947d5e7-04de-48ab-8d34-f4dc30c19ad3" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfAntennas_39HzNd1Dv8guqqzDmmmPhJ/DocProperty.xml:2]: No need for change

File: IFC4x3/Properties/n/NumberOfAntennas_3pcohitgD84wYzTL$rzTOJ/DocProperty.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfAntennas_3pcohitgD84wYzTLrzTOJ" Name="NumberOfAntennas" UniqueId="f39b2aec-dea3-4813-a8bd-755ff5f5d613" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfAntennas_3pcohitgD84wYzTL$rzTOJ/DocProperty.xml:2]: Already have replied too much :)

File: IFC4x3/PropertyEnumerations/r/PEnum_RemoteUnitConnectionType.xml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_RemoteUnitConnectionType" Name="PEnum_RemoteUnitConnectionType" UniqueId="960f2795-0ae4-4243-80ef-119be5b903da">
+	<Constants>
+		<DocPropertyConstant xsi:nil="true" href="CHAIN_3so8ETHqLFfRIPoztV5GAO" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/PropertyEnumerations/r/PEnum_RemoteUnitConnectionType.xml]: All these redundant constants should have been solved

File: IFC4x3/Properties/a/AntennaType_2PysPrqKj3aPWcff71DZsA/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="AntennaType_2PysPrqKj3aPWcff71DZsA" Name="AntennaType" UniqueId="99f36675-d14b-4391-9826-a691c1363d8a" />
+
```
Comment (larswik) [inline IFC4x3/Properties/a/AntennaType_2PysPrqKj3aPWcff71DZsA/DocProperty.xml:3]: PrimaryDataType still missing.

File: IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfInterfaces_3jLpNUj92SQ9VcAUpjI_k" Name="NumberOfInterfaces" UniqueId="ffb55cd7-7ad2-4271-a25f-98a7b3b52fae" PropertyType="p_tablevalue" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/DocProperty.xml]: The redundant one is removed

File: IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/Documentation.md [hunk 9]
```diff
@@ -0,0 +1 @@
+Indicate the types of interfaces and numbers of each type of them of the device.
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfInterfaces_3$jLpNUj92SQ9VcAUpjI_k/Documentation.md]: see above

File: IFC4x3/PropertyConstants/c/CHAIN_3so8ETHqLFfRIPoztV5GAO.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="CHAIN_3so8ETHqLFfRIPoztV5GAO" Name="CHAIN" UniqueId="f6c8839d-4745-4fa5-b499-cbdddf150298" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/PropertyConstants/c/CHAIN_3so8ETHqLFfRIPoztV5GAO.xml]: The duplicated one is removed.

**Review by larswik — APPROVED** [review]

**Review by SergejMuhic — COMMENTED** [review]
Additionally to the comments:
- constants missing documentation
- enumerations missing documentation

File: IFC4x3/Properties/n/NumberOfEmergencyTransceivers_2RrzYcZZf2sO3dp5bbzVgl/DocProperty.xml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfEmergencyTransceivers_2RrzYcZZf2sO3dp5bbzVgl" Name="NumberOfEmergencyTransceivers" UniqueId="9bd7d8a6-8e3a-42d9-80e7-cc5965f5faaf" PrimaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfEmergencyTransceivers_2RrzYcZZf2sO3dp5bbzVgl/DocProperty.xml:2]: No. Will not be modelled in that detail.

File: IFC4x3/Properties/b/BandWidth_3qGbr0Q199cuUUYiO3nY0i/DocProperty.xml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="BandWidth_3qGbr0Q199cuUUYiO3nY0i" Name="BandWidth" UniqueId="f4425d40-6812-499b-879e-8ac603c6202c" PrimaryDataType="IfcFrequencyMeasure" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/b/BandWidth_3qGbr0Q199cuUUYiO3nY0i/DocProperty.xml:2]: Yes, double checked with domain experts, it should be Hz. This is the definition of Band Width in signal processing: https://en.wikipedia.org/wiki/Bandwidth_(signal_processing)

File: IFC4x3/Properties/n/NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf/DocProperty.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf" Name="NumberOfInterfaces" UniqueId="cd03a478-4846-43ab-833d-4727038848e9" PropertyType="p_tablevalue" PrimaryDataType="IfcLabel" SecondaryDataType="IfcInteger" />
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/n/NumberOfInterfaces_3D0wHuI4P3guCzHoS3Y4Zf/DocProperty.xml:2]: Same. BBU is just a device. This property is used to same how many interfaces it has.

File: IFC4x3/Properties/t/TransmittedBandwidth_2x5FCzOXX65wSk8PZymLiB/Documentation.md [hunk 14]
```diff
@@ -0,0 +1 @@
+Indicates the transmitted bandwidth of the master unit.
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/t/TransmittedBandwidth_2x5FCzOXX65wSk8PZymLiB/Documentation.md:1]: Double checked again. It should also be Hz, same with BandWidth.

[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Properties/t/TransmittedBandwidth_2x5FCzOXX65wSk8PZymLiB/Documentation.md:1]: Resolved: https://github.com/bSI-InfraRoom/IFC-Specification/pull/158/commits/6517df5dcde9580e8fafc2cb2dd05c7a8b4e42ff

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_MobileTeleCommunicationsApplianceTypeRemoteRadioUnit/Documentation.md [hunk 15]
```diff
@@ -0,0 +1 @@
+Properties common to the definition of all occurrences of  _IfcMobileTelecommunicationsAppliance_ and types of _IfcMobileTelecommunicationsApplianceType_ with the predefined type set to REMOTERADIOUNIT.
```
[the remark this answers is not in the document]
Comment (czapplitec) [reply IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRailDomain/PropertySets/Pset_MobileTeleCommunicationsApplianceTypeRemoteRadioUnit/Documentation.md:1]: OK, I got the point. Resolved: https://github.com/bSI-InfraRoom/IFC-Specification/pull/158/commits/6517df5dcde9580e8fafc2cb2dd05c7a8b4e42ff

**Pull request merged**